docs/en/ingest-management/processors/processor-communityid.asciidoc

```diff
@@ -0,0 +1,77 @@
+[[community_id-processor]]
+= Community ID Network Flow Hash
+
+++++
+<titleabbrev>community_id</titleabbrev>
+++++
+
+The `community_id` processor computes a network flow hash according to the
+https://github.com/corelight/community-id-spec[Community ID Flow Hash
+specification].
+
+The flow hash is useful for correlating all network events related to a
+single flow. For example, you can filter on a community ID value and you might
+get back the Netflow records from multiple collectors and layer 7 protocol
+records from the Network Packet Capture integration.
+
+By default the processor is configured to read the flow parameters from the
+appropriate Elastic Common Schema (ECS) fields. If you are processing ECS data,
+no parameters are required. 
+
+[discrete]
+== Examples
+
+[source,yaml]
+----
+  - community_id:
+----
+
+If the data does not conform to ECS, you can customize the field names that the
+processor reads from. You can also change the target field that the computed
+hash is written to. For example:
+
+[source,yaml]
+----
+  - community_id:
+      fields:
+        source_ip: my_source_ip
+        source_port: my_source_port
+        destination_ip: my_dest_ip
+        destination_port: my_dest_port
+        iana_number: my_iana_number
+        transport: my_transport
+        icmp_type: my_icmp_type
+        icmp_code: my_icmp_code
+      target: network.community_id
+----
+
+If the necessary fields are not present in the event, the processor silently
+continues without adding the target field.
+
+[discrete]
+== Configuration settings
+
+[options="header"]
+|===
+| Name | Required | Default | Description
+
+| `fields`
+| No
+|
+| Field names that the processor reads from.
+
+| `target`
+| No
+|
+| Field that the computed hash is written to.
+
+| `seed`
+| No
+|
+| 16-bit unsigned integer that gets incorporated into all generated hashes.
+
+|===
+
+//REVIEWERS: Original topic did not provide descriptions of each config setting.
```

Comment:
Need input from reviewers on this question.

Reply:
I can't really provide a good description of several of these. @andrewkroh looks like you wrote this processor and may have better info than me.

Reply:
You can use the Elasticsearch docs of community_id to get descriptions. The usage example above on line 35 list all the available `fields.*` options (e.g. `source_ip`, `source_port`).

https://www.elastic.co/guide/en/elasticsearch/reference/master/community-id-processor.html